packages/jupyter/src/output.tsx

```diff
@@ -37,43 +39,42 @@ export function allOutputsAreSafe(
   }, true);
 }
 
-export function JupyterOutput({
-  outputId,
-  identifier,
-  data,
-  align,
-  className,
-}: {
-  outputId: string;
-  identifier?: string;
-  data: MinifiedOutput[];
-  align?: 'left' | 'center' | 'right';
-  className?: string;
-}) {
-  const { ready } = useCellExecution(outputId);
-  const outputs: MinifiedOutput[] = data;
+export function Output({ node }: { node: GenericNode }) {
+  const { outputsId, allSafe } = useOutputsContext();
+  const { ready } = useCellExecution(outputsId);
+  const outputs = useMemo(() => [node.jupyter_data], [node]);
+  return allSafe && !ready ? (
+    <SafeOutputs keyStub={outputsId} outputs={outputs} />
+  ) : (
+    <JupyterOutputs id={outputsId} outputs={outputs} />
+  );
+}
+
+export function Outputs({ node }: { node: GenericNode }) {
+  const className = classNames({ hidden: node.visibility === 'remove' });
+  const { children, identifier, align } = node;
+  const outputsId = node.id ?? node.key;
+  const cellExecutionContext = useCellExecution(outputsId);
+  const { ready } = cellExecutionContext;
+
+  const outputs: MinifiedOutput[] = useMemo(
+    () => selectAll('output', node).map((child) => (child as any).jupyter_data),
+    [children],
+  );
   const allSafe = useMemo(
```

Comment:
This will likely change as this PR evolves. For now, I'm preserving as much original behaviour as possible in order to make the PR small and easy to reason about.